lib/contentful_middleman/tools/backup.rb

```diff
@@ -0,0 +1,84 @@
+require 'fileutils'
+
+module ContentfulMiddleman
+  module Tools
+    class NullBackup
+      def restore; end
+      def destroy; end
+    end
+
+    class Backup
+
+      class << self
+        def basepath
+          ::File.join ENV["MM_ROOT"], ".tmp", "backups"
+        end
+
+        def ensure_backup_path!
+          return if ::File.exists? basepath
+
+          FileUtils.mkdir_p basepath
+        end
+      end
+
+      ensure_backup_path!
+
+      def initialize(name, source)
+        @name   = name
+        @source = source
+
+        FileUtils.mkdir(path)
+        #FileUtils.cp(all_files_in_path(source), path)
+        FileUtils.mv(source, path)
+      end
+
+
+      def restore
+        FileUtils.rm_rf(@source)
+        FileUtils.mv(path, @source)
+      end
+
+      def destroy
+        FileUtils.rm_rf(path)
+      end
+
+      private
+      def path
+        ::File.join self.class.basepath, name_and_date
+      end
+
+      def all_files_in_path(path)
+        Dir.glob(::File.join(path, "*"))
+      end
+
+      def name_and_date
+        @name_and_date ||= "#{@name}-#{Time.now.strftime("%Y%m%d%H%M%S")}"
+      end
+
+      module InstanceMethods
+        def do_with_backup(backup_name, path_to_backup)
+          backup = create_backup backup_name,  path_to_backup
+
+          begin
+            yield
+          rescue StandardError => e
+            backup.restore
+            raise e
+          ensure
+            backup.destroy
```

Comment:
Now we delete the backup only if there was no problem while executing the block or if the backup has been properly restored. Done here https://github.com/madtrick/contentful_middleman/commit/9ff953d0d3b97ae1f42e39beee0facfc383e3542

It isn't a bulletproof solution but we aren't creating here a backup solution 😝